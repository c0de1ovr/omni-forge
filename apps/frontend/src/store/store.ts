import { createStore } from 'zustand/vanilla';

import {
  createCounterSlice,
  type CounterSlice,
} from './features/counter/counter.slice';

export type AppStore = CounterSlice;

export const appStore = createStore<AppStore>()((...args) => ({
  ...createCounterSlice(...args),
}));
