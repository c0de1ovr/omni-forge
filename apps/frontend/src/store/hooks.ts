import { useContext } from 'react';
import { useStore } from 'zustand';

import { AppStoreContext } from './provider';
import { type AppStore } from './store';

export const useAppStore = <T>(selector: (store: AppStore) => T): T => {
  const appStoreContext = useContext(AppStoreContext);

  if (!appStoreContext) {
    throw new Error(`appStoreContext must be use within AppStoreProvider`);
  }

  return useStore(appStoreContext, selector);
};
