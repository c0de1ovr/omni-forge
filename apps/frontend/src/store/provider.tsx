'use client';
import { type ReactNode, useRef, type FC } from 'react';
import { Provider } from 'react-redux';

import { makeStore, type AppStore } from './store';

export const StoreProvider: FC<{ children: ReactNode }> = ({ children }) => {
  const storeRef = useRef<AppStore>();
  if (!storeRef.current) {
    // Create the store instance the first time this renders
    storeRef.current = makeStore();
  }

  return <Provider store={storeRef.current}>{children}</Provider>;
};
