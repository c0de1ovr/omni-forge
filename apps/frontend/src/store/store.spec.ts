import { configureStore } from '@reduxjs/toolkit';

import { makeStore } from './store';

jest.mock('@reduxjs/toolkit');
jest.mock('./features/counter/counter.slice', () => 'mocked-counter-slice');

describe('store', () => {
  it('should apply slices correctly', () => {
    makeStore();
    expect(configureStore).toHaveBeenCalledTimes(1);
    expect(configureStore).toHaveBeenCalledWith({
      reducer: { counter: 'mocked-counter-slice' },
    });
  });
});
