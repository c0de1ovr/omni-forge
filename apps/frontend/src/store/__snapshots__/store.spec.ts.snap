// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store should apply slices correctly 1`] = `
{
  "decrement": [Function],
  "increment": [Function],
  "incrementByAmount": [Function],
  "value": 0,
}
`;

exports[`store should apply slices correctly 2`] = `
{
  "decrement": [Function],
  "increment": [Function],
  "incrementByAmount": [Function],
  "value": 0,
}
`;
